agent/metrics.go

```diff
@@ -24,6 +93,33 @@ func collectMetrics() []agentsdk.AgentMetric {
 			Value: float64(m.Value()),
 		})
 	}
+
+	// Agent metrics
+	metricFamilies, err := a.prometheusRegistry.Gather()
+	if err != nil {
+		a.logger.Error(ctx, "can't gather agent metrics", slog.Error(err))
+		return collected
+	}
+
+	for _, metricFamily := range metricFamilies {
+		for _, metric := range metricFamily.GetMetric() {
+			if metric.Counter != nil {
+				collected = append(collected, agentsdk.AgentMetric{
+					Name:  metricFamily.GetName(),
```

Comment:
Oh man, it's a real shame we never thought to include labels on metrics in the first place.

I think we should.  They're just too useful to ignore and we publish on the coderd side with prometheus.

Reply:
Let me share some background - in the [original implementation](https://github.com/coder/coder/pull/7183) I focused on passing the magicsock metrics that are pretty flat (no vectors, no labels). I didn't have a reason to implement an overkill with unused labels (KISS).

This is the first time on the agent side that requires support for metric vectors (and labels). As you noticed, it means that we need to improve the receiver side too (coderd).

I'm not quite sure if I should do this as part of this PR, or as a separate story, and refactor metrics to use labels.

Reply:
Added support for aggregating metrics by labels.